packages/commonjs/src/transform-commonjs.js

```diff
@@ -165,6 +186,23 @@ export default function transformCommonjs(
             skippedNodes.add(node.callee);
             const usesReturnValue = parent.type !== 'ExpressionStatement';
 
+            let canConvertRequire = true;
+            let shouldRemoveRequireStatement = false;
+
+            if (tryCatchState.depth > 0) {
+              let mode = getIgnoredTryCatchRequireStatementMode(node, ignoreTryCatch);
```

Comment:
Why not extract most of the logic here into the function? E.g.

```js
({canConvertRequire, shouldRemoveRequireStatement}) = getTryCatchRequireStatementHandling(node, ignoreTryCatch)
```

Reply:
Or even more efficient, replace the `ignoreTryCatch` function directly with a function that returns the information we actually need, i.e. `canConvertRequire` and `shouldRemoveRequireStatement` for a given id. And instead of a separate function that extracts the id from the node, directly call it with `node.arguments[0].value`?